### ISSUE TITLE
Remove payment profile ID from usermeta when cancelling subscription

### ISSUE THREAD

**Issue by nosegraze** [restrictcontentpro/restrict-content-pro#1375]
Currently we never remove the payment profile ID unless we're replacing it with another. Would there be any downsides to removing the ID whenever the associated subscription is cancelled?

It can be a bit confusing to see a payment profile ID in a profile if it's no longer actually associated with the account. Here's an example along with a problem a customer ran into:

1. User signed up for a recurring subscription and payment profile ID was filled out. Expiration set to `2017-08-03`.
1. The membership was cancelled to stop the automatic billing. Old payment profile ID was still set though.
1. The user later renewed the membership manually (this time non-recurring). Expiration extended to `2018-08-03`.
1. But on August 3rd 2017 (original expiration date from step 1), PayPal sent a `subscr_eot` IPN for the cancelled plan and since the old payment profile ID was still saved in the account, RCP expired the account even though the expiration date had since been extended.

tl;dr: Membership was incorrectly expired early.

It can also perhaps be confusing if someone moves from a paid recurring plan to a free plan but still has the payment profile ID filled out, which is now no longer applicable. I think this would also be the case if someone was on a paid recurring plan, then upgraded to a different plan that's non-recurring. The old payment profile ID would still be there.

**Comment by mindctrl** [comment]
I can't think of a reason we should keep it. We could log it in a member note for a historical record to make it easier for someone to reference in the future if needed.

**Comment by nosegraze** [comment]
I like that idea of logging it in the member notes.

**Comment by pippinsplugins** [comment]
As long as it is noted somewhere I'm fine with removing it. It can be
really valuable for looking up history of an account

**Comment by nosegraze** [comment]
One potential downside: If we remove the Stripe customer ID then I believe this would create a new Stripe customer record if/when a user cancels and then signs up again later. A little annoying on the Stripe record-keeping side.

**Comment by mindctrl** [comment]
If we remove the payment profile ID when the sub is cancelled, it makes it harder to build an accurate picture when we migrate subscriptions to a custom table. In this context, as long as we only remove it when the member changes to another level, I think it will be ok.

**Comment by nosegraze** [comment]
PR: https://github.com/restrictcontentpro/restrict-content-pro/pull/1582

- I logged all changes to the payment profile ID in user notes.
- I removed the payment profile ID when moving to a free subscription (fully discounted or actually free level).
- I removed the payment profile ID when making a payment via the Manual gateway.
- I removed the payment profile ID when a one-time transaction is successful in Authorize.net, PayPal Standard, and PayPal Express.

I decided to leave them there for the other gateways since they signify a "customer ID" rather than a "subscription ID" and might be useful for more easily looking up the customer in Stripe/Braintree.

**Comment by mindctrl** [comment]
Going to punt this to 3.0.1 so we can do this after we migrate subscriptions per https://github.com/restrictcontentpro/restrict-content-pro/issues/1375#issuecomment-344313916.

**Comment by nosegraze** [comment]
Got another report of the exact four steps I reported before. This customer has had several members expiring because of this.
